src/client/general/protocol.md

```diff
@@ -0,0 +1,58 @@
+# The Wayland protocol
+
+The Wayland protocol can be described as an asynchronous, object oriented protocol. This means clients and
+the compositor send and receive messages in order to build an abstract state machine made of objects.
+Wayland as a protocol is also asynchronous in the sense that a message will not complete immediately.
+
+## Interfaces
+
+The Wayland protocol is defined in terms of interfaces. Interfaces describe the signature of all messages
+that may be sent or received by an object. An interface when instantiated becomes a protocol object. The
+protocol object may accept and send messages that were defined in the interface.
```

Comment:
Okay, this is very nitpicky, but from what I've seen this can be a source of confusion. I think we should avoid terminology like "the object sends a message", because that leaves open the question of "to what is the message sent", which does not really have a good answer because it's not the right mental image.

Rather, I think we should say that the client and server both send messages to the other end, and that each message is associated with a protocol object.